### ISSUE TITLE
[Math Processing Error]

### ISSUE THREAD

**Issue by karna77** [mathjax/MathJax#1032]
Hello,

The below math has rendered perfectly in firefox and chrome yesterday, but today we are getting 'Math processing error'. We are unable to find the issue behind this. Could you please shed light on this?

```
<m:math display='block'><m:mtable columnalign='left'><m:mtr><m:mtd><m:mtable><m:mtr><m:mtd><m:mrow><m:mtable frame='solid'><m:mrow><m:mtext>Multiples&#x00A0;of&#x00A0;</m:mtext><m:mn>2</m:mn></m:mrow></m:mtable></m:mrow></m:mtd><m:mtd><m:mrow><m:mspace width="1em"/><m:mtable frame='solid'><m:mrow><m:mtext>Multiples&#x00A0;of&#x00A0;</m:mtext><m:mn>3</m:mn></m:mrow></m:mtable></m:mrow></m:mtd></m:mtr><m:mtr><m:mtd><m:mspace width="0.4em"/><m:mo stretchy="false">&#x2193;</m:mo></m:mtd><m:mtd><m:mspace width="1.5em"/><m:mo stretchy="false">&#x2193;</m:mo></m:mtd></m:mtr></m:mtable></m:mtd></m:mtr><m:mtr><m:mtd><m:mtable columnalign='left'><m:mtr columnalign='left'><m:mtd columnalign='left'><m:mrow><m:mn>2</m:mn><m:mo>&#x22C5;</m:mo><m:mn>1</m:mn><m:mo>=</m:mo><m:mspace width="0.5em"/><m:mn>2</m:mn><m:mspace width="4em"/></m:mrow></m:mtd><m:mtd columnalign='left'><m:mrow><m:mn>3</m:mn><m:mo>&#x22C5;</m:mo><m:mn>1</m:mn><m:mo>=</m:mo><m:mspace width="0.5em"/><m:mn>3</m:mn></m:mrow></m:mtd></m:mtr><m:mtr columnalign='left'><m:mtd columnalign='left'><m:mrow><m:mn>2</m:mn><m:mo>&#x22C5;</m:mo><m:mn>2</m:mn><m:mo>=</m:mo><m:mspace width="0.5em"/><m:mn>4</m:mn></m:mrow></m:mtd><m:mtd columnalign='left'><m:mrow><m:mn>3</m:mn><m:mo>&#x22C5;</m:mo><m:mn>2</m:mn><m:mo>=</m:mo><m:mspace width="0.5em"/><m:mn>6</m:mn></m:mrow></m:mtd></m:mtr><m:mtr columnalign='left'><m:mtd columnalign='left'><m:mrow><m:mn>2</m:mn><m:mo>&#x22C5;</m:mo><m:mn>3</m:mn><m:mo>=</m:mo><m:mspace width="0.5em"/><m:mn>6</m:mn></m:mrow></m:mtd><m:mtd columnalign='left'><m:mrow><m:mn>3</m:mn><m:mo>&#x22C5;</m:mo><m:mn>3</m:mn><m:mo>=</m:mo><m:mspace width="0.5em"/><m:mn>9</m:mn></m:mrow></m:mtd></m:mtr><m:mtr columnalign='left'><m:mtd columnalign='left'><m:mrow><m:mn>2</m:mn><m:mo>&#x22C5;</m:mo><m:mn>4</m:mn><m:mo>=</m:mo><m:mspace width="0.5em"/><m:mn>8</m:mn></m:mrow></m:mtd><m:mtd columnalign='left'><m:mrow><m:mn>3</m:mn><m:mo>&#x22C5;</m:mo><m:mn>4</m:mn><m:mo>=</m:mo><m:mn>12</m:mn></m:mrow></m:mtd></m:mtr><m:mtr columnalign='left'><m:mtd columnalign='left'><m:mrow><m:mn>2</m:mn><m:mo>&#x22C5;</m:mo><m:mn>5</m:mn><m:mo>=</m:mo><m:mn>10</m:mn></m:mrow></m:mtd><m:mtd columnalign='left'><m:mrow><m:mn>3</m:mn><m:mo>&#x22C5;</m:mo><m:mn>5</m:mn><m:mo>=</m:mo><m:mn>15</m:mn></m:mrow></m:mtd></m:mtr><m:mtr columnalign='left'><m:mtd columnalign='left'><m:mrow><m:mn>2</m:mn><m:mo>&#x22C5;</m:mo><m:mn>6</m:mn><m:mo>=</m:mo><m:mn>12</m:mn></m:mrow></m:mtd><m:mtd columnalign='left'><m:mrow><m:mn>3</m:mn><m:mo>&#x22C5;</m:mo><m:mn>6</m:mn><m:mo>=</m:mo><m:mn>18</m:mn></m:mrow></m:mtd></m:mtr></m:mtable></m:mtd></m:mtr></m:mtable></m:math>
```

Looking forward to hear from you.

Regards,
Karnan S
karnan.subramanian@spi-global.com

**Comment by pkra** [comment]
Please provide a live sample page (e.g., using jsbin etc) that exhibits the problem.

But a wild guess would be that you are using the CDN's `latest` branch and given yesterday's  release of 2.5 you now have mixed versions of MathJax in your browser cache. See if it disappears when you clear your browser cache.

**Comment by CharlesAnifowose** [comment]
We've just started facing this in the past 12 hours as well. I was able to reproduce it when I first opened the page, but when I refreshed, everything started displaying correctly. Not everyone who experiences this will know to do this though.  

We're pointing to: 
`//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML`
which we suspect may be the cause.

**Comment by dpvc** [comment]
The MathML you site also renders for me.  I suspect Peter is right, that it is a caching issue causing you to get a mixture of files from 2.4 and 2.5.  Clearing your browser cache and restarting the browser should clear that up.  If not, can you temporarily change the URL where you load MathJax to `//cdn.mathjax.org/mathjax/2.5-latest` and see if that helps?

**Comment by pkra** [comment]
@dpvc re the OP. I could reproduce an issue which seems to happen with the Fast Preview enabled but not without it. The error reported is `Error: I is null` -- see http://codepen.io/pkra/pen/ZYXYmE.

**Comment by dpvc** [comment]
@pkra, OK, if I use the CommonHTML output, I do get that result.

The problem is that the original MathML is malformed.  For example, it contains:

    <m:mtable frame='solid'>
      <m:mrow>
        <m:mtext>Multiples&#x00A0;of&#x00A0;</m:mtext>
        <m:mn>2</m:mn>
      </m:mrow>
    </m:mtable>

But the children of `<mtable>` should be `<mtr>` not `<mrow>`.  If you add in the missing `<mtr><mtd>` and `</mtd></mtr>` in both tables used for the labels at the top, then CommonHTML will display it properly.

I will look into how best to resolve the situation.

**Comment by dpvc** [comment]
I should probably have the CHTML-preview extension trap errors rather then prevent the second pass from running.

**Comment by pkra** [comment]
Doh -- I didn't spot that. It's duplicate of #1006 then?

**Comment by dpvc** [comment]
Yes, they both are due to the improper mtable tree.

**Comment by pkra** [comment]
Closing this as duplicate then?

**Comment by dpvc** [comment]
Sure.

**Comment by pkra** [comment]
=> Closed; duplicate of #1006
